### PR TITLE
Handle `*-deps` jobs in homebrew/core

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
Needed for https://github.com/Homebrew/homebrew-core/pull/125595

### CODE REVIEW

**Review by carlocab — COMMENTED** [review]
This would be my guess too for how it would work, but it would be good to get @Bo98 to confirm.

File: src/github_watcher.rb [hunk 1]
```diff
@@ -196,7 +196,7 @@ def timeout_queued_and_deployed
       next if job.github_state != :queued
       next if job.orka_vm_id.nil?
 
-      run_key = "#{job.run_id}-#{job.run_attempt}"
+      run_key = "#{job.run_id}-#{job.run_attempt}#{job.deps}"
```
Comment (Bo98) [inline src/github_watcher.rb:199]: I don't think this is needed - this only needs to check the GitHub API once per run ID & attempt combo as it doesn't consider individual steps and only the larger workflow run (this API: https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#get-a-workflow-run-attempt).

Maybe could refine it to be per step later, though it's non-trivial and all this is only necesary if the GitHub webhooks suffer a complete loss of events.

**Review by Bo98 — COMMENTED** [review]
Looks good - thanks!

File: src/github_watcher.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply src/github_watcher.rb:199]: @Bo98 Ok so only this line/change isn't needed but the `NAME_REGEX` below looks good/ok?

Comment (Bo98) [reply src/github_watcher.rb:199]: Yes

Comment (MikeMcQuaid) [reply src/github_watcher.rb]: @Bo98 Great, thanks!

**Pull request merged**